src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -424,7 +424,7 @@ private Task ExtractToFileInternalAsync(string filePath, string? linkTargetPath,
 
         private void CreateNonRegularFile(string filePath, string? linkTargetPath)
         {
-            Debug.Assert(EntryType is not TarEntryType.RegularFile or TarEntryType.V7RegularFile or TarEntryType.ContiguousFile);
+            Debug.Assert(EntryType is not (TarEntryType.RegularFile or TarEntryType.V7RegularFile or TarEntryType.ContiguousFile));
```

Comment:
nit: alternative fix is what we are using in number of places (including this file):
```suggestion
            Debug.Assert(EntryType is not TarEntryType.RegularFile and not TarEntryType.V7RegularFile or TarEntryType.ContiguousFile);
```